packages/pelagos/src/treeView/TreeNode.js

```diff
@@ -0,0 +1,197 @@
+import {useCallback, useContext, useRef} from 'react';
+import PropTypes from 'prop-types';
+import {faCaretRight} from '@fortawesome/free-solid-svg-icons';
+
+import SvgIcon from '../components/SvgIcon';
+
+import TreeViewContext from './TreeViewContext';
+import TreeNodeContext from './TreeNodeContext';
+
+/** A node in a TreeView. */
+const TreeNode = ({id, labelClassName, label, icon, expanded, loading, children, onToggle, ...props}) => {
+	const nodeRef = useRef(null);
+	const {selected, focused, setFocused, onSelect} = useContext(TreeViewContext);
+	const {level, padding: parentPadding, hasIcon} = useContext(TreeNodeContext);
+	const leaf = expanded === undefined;
+	const padding = parentPadding + 16 + (leaf ? 24 : 0) + (hasIcon ? 8 : 0);
+	const nodeContextValue = {level: level + 1, padding, hasIcon: !!icon};
+
+	const handleClick = useCallback(
+		(event) => {
+			event.stopPropagation();
+			onSelect?.(id);
+			setFocused(id);
+		},
+		[id, onSelect, setFocused]
+	);
+	const handleToggleClick = useCallback(() => {
+		onToggle?.(id, !expanded);
```

Comment:
In my own testing, the node is selected and toggled when clicking to toggle. Is that correct? I'd expect clicking to toggle to only toggle.

Reply:
I'm following the behaviour in Carbon, check https://react.carbondesignsystem.com/?path=/story/components-treeview--default

Reply:
GitHub's file tree has the click behaviour I'd expect (and what CDS seemed to specify in the page I linked above), except that they auto-expand when selecting parent nodes.